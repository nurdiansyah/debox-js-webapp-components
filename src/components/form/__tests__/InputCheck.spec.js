import React from 'react'

import {Form, InputCheck, getFormBlock, getInputCheckBlock} from '../index'

const values = [
  {label: 'Mangga', value: 'mangga'},
  {label: 'Mangga', value: 'mangga'},
  {id: '2', label: 'Jeruk', value: 'jeruk'},
  {id: '3', label: 'Apel', value: 'apel', checked: true}
]

describe('form/Radio', () => {
  it('should running no error', () => {
    const formBlock = getFormBlock({})
    const inputCheckBlock = getInputCheckBlock({})
    const wrapper = render(
      <Form block={formBlock}>
        <InputCheck block={inputCheckBlock} name="buah" values={values} label="label" multiple />
      </Form>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
