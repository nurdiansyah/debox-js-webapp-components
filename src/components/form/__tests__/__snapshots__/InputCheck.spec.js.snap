// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form/Radio should running no error 1`] = `
<form
  class="form"
>
  <div
    class="form__control"
  >
    <div
      class="input-check "
    >
      <label
        class="input-check__label"
        for="buah2"
      >
        <input
          class="input-check__checkbox"
          id="buah2"
          type="checkbox"
          value="mangga"
        />
        Mangga
      </label>
      <label
        class="input-check__label"
        for="buah3"
      >
        <input
          class="input-check__checkbox"
          id="buah3"
          type="checkbox"
          value="mangga"
        />
        Mangga
      </label>
      <label
        class="input-check__label"
        for="2"
      >
        <input
          class="input-check__checkbox"
          id="2"
          type="checkbox"
          value="jeruk"
        />
        Jeruk
      </label>
      <label
        class="input-check__label"
        for="3"
      >
        <input
          class="input-check__checkbox"
          id="3"
          type="checkbox"
          value="apel"
        />
        Apel
      </label>
    </div>
  </div>
</form>
`;
